src/main/java/owen/Owen.java

```diff
@@ -0,0 +1,172 @@
+package owen;
+
+import owen.command.Command;
+import owen.task.TaskList;
+
+/**
+ * Owen is a personal assistant chatbot.
+ * As Owen is an immutable class, modifications will return a copy
+ * with updated internal state.
+ */
+public class Owen implements Chatbot {
+    private static final String STORAGE_PATH = "data/owen.txt";
+
+    private final boolean isRunning;
+    private final String latestResponse;
+    private final TaskList taskList;
+    private final Storage storage;
+
+    /**
+     * Creates an Owen chatbot with a hello response.
+     */
+    public Owen() {
+        this.isRunning = true;
+
+        StringBuilder stringBuilder = new StringBuilder();
+        String logo = ""
+                + " /\\_/\\ \n"
+                + "((OvO))\n"
+                + "():::()\n"
+                + " VV-VV \n";
+        stringBuilder.append(logo);
+        stringBuilder.append("\nHello I am Owen the Owl!");
+        this.latestResponse = stringBuilder.toString();
+
+        this.storage = new Storage(STORAGE_PATH);
+        this.taskList = this.storage.readTaskList();
+    }
+
+    private Owen(boolean isRunning, String latestResponse, TaskList taskList, Storage storage) {
+        this.isRunning = isRunning;
+        this.latestResponse = latestResponse;
+        this.taskList = taskList;
+        this.storage = storage;
+    }
+
+    @Override
+    public Owen shutdown() {
+        String shutdownResponse = "Bye. Hope to see you again soon!";
+        return new Owen(false, shutdownResponse, this.taskList, this.storage);
+    }
+
+    @Override
+    public boolean isRunning() {
+        return this.isRunning;
+    }
+
+    @Override
+    public String getResponse() {
+        return this.latestResponse;
+    }
+
+    @Override
+    public Owen parseCommand(String commandString) {
+        try {
+            Command command = Parser.parseCommand(commandString);
+
+            switch (command.getType()) {
+            case TODO:
+            case EVENT:
+            case DEADLINE:
+                return this.addTask(command.getOriginal());
+            case LIST:
+                return this.listTasks();
+            case DONE:
+                return this.doneTask(parseTaskNumber(command.getArgs()));
+            case DELETE:
+                return this.deleteTask(parseTaskNumber(command.getArgs()));
+            case FIND:
+                return this.findTask(command.getArgs());
+            case BYE:
+                return this.shutdown();
+            default:
+                throw new OwenException("I'm sorry, but I don't know what that means...");
+            }
+        } catch (OwenException exception) {
+            String exceptionResponse = exception.getMessage();
+            return new Owen(this.isRunning, exceptionResponse, this.taskList, this.storage);
+        }
+    }
+
+    private static int parseTaskNumber(String taskNumber) throws OwenException {
```

Comment:
I am thinking for this part, perhaps we can give it to the Parser to do? Just my thoughts :)